### ISSUE TITLE
 project: required field is not set when creating google_service_account_key

### ISSUE THREAD

**Issue by petems** [terraform-providers/terraform-provider-google#1683]
### Terraform Version
Terraform v0.11.7
+ provider.google v1.14.0
+ provider.random v1.3.1

### Affected Resource(s)
- google_service_account
- google_service_account_key

### Terraform Configuration Files
```hcl
variable "billing_account" {}

variable "org_id" {
}

provider "google" {
  version = "1.14.0"
}

resource "random_id" "id" {
 byte_length = 4
 prefix      = "vault-gcp-demo-"
}

resource "google_project" "vault_gcp_demo" {
 name            = "vault-gcp-demo"
 project_id      = "${random_id.id.hex}"
 billing_account = "${var.billing_account}"
 org_id          = "${var.org_id}"
}

resource "google_project_services" "vault_gcp_demo_services" {
 project = "${google_project.vault_gcp_demo.project_id}"
 services = [
   "oslogin.googleapis.com",
   "compute.googleapis.com",
   "iamcredentials.googleapis.com"
 ]
}

output "project_id" {
 value = "${google_project.vault_gcp_demo.project_id}"
}

resource "google_service_account" "vault_auth_checker" {
  project      = "${google_project.vault_gcp_demo.project_id}"
  account_id   = "vault-auth-checker"
  display_name = "Vault Auth Checker"
}

resource "google_project_iam_policy" "vault_policy" {
  project     = "${google_project.vault_gcp_demo.project_id}"
  policy_data = "${data.google_iam_policy.vault_policy.policy_data}"
}

data "google_iam_policy" "vault_policy" {

  binding {
    role = "roles/compute.viewer"

    members = [
      "serviceAccount:${google_service_account.vault_auth_checker.email}",
    ]
  }

  binding {

    role = "roles/iam.securityReviewer"

    members = [
      "serviceAccount:${google_service_account.vault_auth_checker.email}",
    ]

  }
}

resource "google_service_account_key" "vault_auth_checker_credentials" {
  service_account_id = "${google_service_account.vault_auth_checker.name}"
  public_key_type = "TYPE_X509_PEM_FILE"
}

output "service_account_credentials" {
  value = "${base64decode(google_service_account_key.vault_auth_checker_credentials.private_key)}"
}
```

### Debug Output
https://gist.github.com/petems/0da5f1f5c900376124ba06c2dac6bb74

### Expected Behavior
Key should have been created, and the key shown.

I followed the docs for creating an account key here: https://www.terraform.io/docs/providers/google/r/google_service_account_key.html

### Actual Behavior
Error stating that the project field is missing:

```
* google_service_account_key.vault_auth_checker_credentials: 1 error(s) occurred:

* google_service_account_key.vault_auth_checker_credentials: project: required field is not set

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

### Steps to Reproduce
1. `terraform apply`

**Comment by petems** [comment]
I'm assuming that I've got the format of the `service_account_id` incorrect, as the docs say 

>  The Service account id of the Key Pair. This can be a string in the format {ACCOUNT} or projects/{PROJECT_ID}/serviceAccounts/{ACCOUNT}, where {ACCOUNT} is the email address or unique id of the service account. If the {ACCOUNT} syntax is used, the project will be inferred from the account.

But I've tried both and neither work:

```
service_account_id = "${google_service_account.vault_auth_checker.email}" # "vault-auth-checker@vault-gcp-demo-f859ab94.iam.gserviceaccount.com"
service_account_id = "${google_service_account.vault_auth_checker.email}" # projects/vault-gcp-demo-f859ab94/serviceAccounts/vault-auth-checker@vault-gcp-demo-f859ab94.iam.gserviceaccount.com
```

**Comment by paddycarver** [comment]
This is a dupe of #1655 and is fixed by #1664. Sorry for the problems!
